### PR TITLE
test rex connect by ip

### PR DESCRIPTION
**Pull request by pondrejk**
Related to #4624 tests TBA

### CODE REVIEW

**Comment by pondrejk** [comment]
I realized the UI part needs some more work so I decided to move it to a separate commit, test results for the cli part:

```
nosetests -v tests/foreman/cli/test_remoteexecution.py:RemoteExecutionTestCase.test_positive_run_default_job_template_by_ip
Run default template on host connected via ip ... ok

----------------------------------------------------------------------
Ran 1 test in 590.237s

OK

nosetests -v tests/foreman/cli/test_remoteexecution.py:RemoteExecutionTestCase.test_positive_run_custom_job_template_by_ip
Run custom template on host connected via ip ... ok

----------------------------------------------------------------------
Ran 1 test in 601.654s

OK

```

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -328,6 +329,75 @@ def test_positive_run_custom_job_template(self):
...
 
+    @tier2
+    def test_positive_run_default_job_template_by_ip(self):
+        """Run default template on host connected via ip
+
+        :id: 811c7747-bec6-4a2d-8e5c-b5045d3fbc0d
+
+        :expectedresults: Verify the job was successfully ran against the host
+        """
+        Settings.set({
```
Comment (oshtaier) [inline tests/foreman/cli/test_remoteexecution.py]: same comment here about `try...finally` block

Comment (pondrejk) [inline tests/foreman/cli/test_remoteexecution.py]: @oshtaier addressed, ok with this form?

**Comment by pondrejk** [comment]
Hello, after some research and discussion with @rplevka, I decided to take a different approach to these tests (using per-host setting instead of global setting). Therefore I put do_not_merge. Also this can help us to get more rex tests running in automation until the problems from #4578 are resolved

**Comment by pondrejk** [comment]
@renzon @oshtaier ready for review again, I enable the remote_execution_connect_by_ip per host, I added more tests -- these are by_ip version of existing tests. Tests:

```
py.test tests/foreman/cli/test_remoteexecution.py -k _by_ip 
========================================= test session starts ==========================================
platform linux2 -- Python 2.7.11, pytest-2.9.2, py-1.4.33, pluggy-0.3.1
rootdir: /home/pondrejk/Documents/robottelo, inifile: 
plugins: xdist-1.15.0, services-1.1.14, html-1.10.0, cov-2.3.1
collected 21 items 
2017-05-31 10:27:44 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_remoteexecution.py .....s

================================== 15 tests deselected by '-k_by_ip' ===================================
========================= 5 passed, 1 skipped, 17 deselected in 569.50 seconds =========================
```

Also added a skip for two tests due to https://bugzilla.redhat.com/show_bug.cgi?id=1451675

**Pull request merged**